File: sdk/swift/Sources/TelecomSDK/PaymentAPI.swift
import Foundation

/// API for payment management
public class PaymentAPI {
    private let client: HTTPClient
    
    public init(client: HTTPClient) {
        self.client = client
    }
    
    /// Create a payment transaction
    public func createTransaction(_ request: CreatePaymentRequest) async throws -> PaymentTransaction {
        return try await client.post(path: "/v1/payments/transactions", body: request)
    }
    
    /// Get a payment transaction by ID
    public func getTransaction(transactionID: String) async throws -> PaymentTransaction {
        return try await client.get(path: "/v1/payments/transactions/\(transactionID)")
    }
    
    /// List payment transactions
    public func listTransactions(
        subscriberID: Int64? = nil,
        status: PaymentStatus? = nil,
        page: Int = 1,
        pageSize: Int = 50
    ) async throws -> SubscriberList {
        var params: [String: String] = [
            "page": "\(page)",
            "page_size": "\(pageSize)"
        ]
        
        if let subscriberID = subscriberID {
            params["subscriber_id"] = "\(subscriberID)"
        }
        if let status = status {
            params["status"] = "\(status)"
        }
        
        return try await client.get(path: "/v1/payments/transactions", params: params)
    }
}

File: sdk/kotlin/src/main/kotlin/com/telecom/PaymentAPI.kt
package com.telecom

/**
 * API for payment management
 */
class PaymentAPI(private val client: HTTPClient) {
    
    /**
     * Create a payment transaction
     */
    suspend fun createTransaction(request: CreatePaymentRequest): PaymentTransaction {
        return client.post("/v1/payments/transactions", request)
    }
    
    /**
     * Get a payment transaction by ID
     */
    suspend fun getTransaction(transactionId: String): PaymentTransaction {
        return client.get("/v1/payments/transactions/$transactionId")
    }
    
    /**
     * List payment transactions
     */
    suspend fun listTransactions(
        subscriberId: Long? = null,
        status: PaymentStatus? = null,
        page: Int = 1,
        pageSize: Int = 50
    ): SubscriberList {
        val params = mutableMapOf(
            "page" to page.toString(),
            "page_size" to pageSize.toString()
        )
        
        subscriberId?.let { params["subscriber_id"] = it.toString() }
        status?.let { params["status"] = it.toString() }
        
        return client.get("/v1/payments/transactions", params)
    }
}
